### ISSUE TITLE
Why is EntityFrameworkCore.DbMigrations project referenced by HttpApi.Host and IdentityServer project?

### ISSUE THREAD

**Issue by hillin** [abpframework/abp#9419]
In the (tiered) startup template, the `EntityFrameworkCore.DbMigrations` project is referenced by both `HttpApi.Host` and `IdentityServer` project. Also the `EntityFrameworkCoreDbMigrationsModule` is depended by the `HttpApiHostModule` and `IdentityServerModule`. However since nothing in `EntityFrameworkCore.DbMigrations` is used in runtime, why would it be referenced by these runtime assemblies?

**Comment by realLiangshiwei** [comment]
See https://github.com/abpframework/abp/issues/8776

**Comment by hillin** [comment]
@realLiangshiwei Yes I'm aware of that issue. In our project we'd prefer to keep a cleaner separation between the DB migration contexts and the runtime DB contexts. We are even considering to merge the DB migration project with the DB migrator - if the DB migration stuff is not required in the runtime, the DB migrator may be one of the only two projects that need it (along with the `EntityFrameworkCore.Tests` project, but we are okay if it references the DB migrator directly). 

So the question being, prior to v4.4, is there any implication that `EntityFrameworkCore.DbMigrations` is referenced by `HttpApi.Host` and `IdentityServer`? Would it be safe to remove these references?

**Comment by realLiangshiwei** [comment]
See: https://community.abp.io/articles/unifying-dbcontexts-for-ef-core-removing-the-ef-core-migrations-project-nsyhrtna

You can remove the migration project.

**Comment by hillin** [comment]
@realLiangshiwei well yes... that article is about the same thing of #8776.
Say if we unify the DB contexts as the article instructed, `HttpApi.Host` and `IdentityServer` now would only need to reference the `EntityFrameworkCore` project, which they already did; but they can still *see* the DB migration stuffs, and their modules are still *depending on* them. However what I'd like to understand is, what would go wrong if they *can't* see and depend on them.

Or I'd rephrase the question as: is `*MigrationsDbContext` needed in the runtime? It is registered in the `*EntityFrameworkCoreDbMigrationsModule`. Would it be safe to remove the following code *in the runtime*?

```c#
public override void ConfigureServices(ServiceConfigurationContext context)
{
    context.Services.AddAbpDbContext<AppMigrationsDbContext>();
}
```

**Comment by realLiangshiwei** [comment]
Please check the official documentation before asking questions
https://docs.abp.io/en/abp/latest/Startup-Templates/Application#entityframeworkcore-dbmigrations-project

**Comment by hillin** [comment]
@realLiangshiwei Could you please shed some light on how that documentation helps on this topic? I'm aware of every line in that documentation, as it says:
>    Notice that the migration DbContext is only used for database migrations and not used on runtime.

But still the `EntityFrameworkCore.DbMigrations` project is referenced by *runtime* projects in the startup template, which is what I'm wondering here.

Even if the reference is unified and removed as addressed by the article above, these non-runtime classes are still *visible* and *depended by* the runtime code.

**Comment by realLiangshiwei** [comment]
Because we want to use `dotnet ef` and  `add-migration` command. long story to short: you can delete it according to this article: https://community.abp.io/articles/unifying-dbcontexts-for-ef-core-removing-the-ef-core-migrations-project-nsyhrtna

**Comment by hillin** [comment]
> Because we want to use `dotnet ef` and `add-migration` command. 

Thanks, that's reasonable.
